colour/colorimetry/spectrum.py

```diff
@@ -2659,3 +2664,75 @@ def sds_and_msds_to_msds(sds):
             tstack(values), shape.range(), labels, strict_labels=strict_labels)
 
     return msds
+
+
+_CACHE_RESHAPED_SDS_AND_MSDS = {}
+
+
+def reshape_sd(sd, shape=SPECTRAL_SHAPE_DEFAULT, method='Align', **kwargs):
```

Comment:
Could this also be achieved via an [LRU cache](https://docs.python.org/3/library/functools.html#functools.lru_cache), instead of a custom caching implementation? Just adding the annotation seems simpler and less error-prone.

Reply:
This is a good question, right now the cache is a simple as it can be, i.e. a dict, because I did not felt like it warranted anything fancier than that. However, with #846, it should be trivial to play and test various style of caches!

Reply:
Oh and as far as using an LRU cache via the decorator, it would not work because so of the arguments are dictionaries, e.g. `extrapolotor_kwargs={'left': 0, 'method': constant}`, and thus they are not hashable (and usable directly)!

Reply:
So there is that option: https://stackoverflow.com/a/53394430/931625 but it is becoming overkill.

We also don't have too much control over the cache like I would like to have with #846.

Another minor thing but also worth noting is that keyword arguments order is important for `lru_cache`, e.g. `foo(a=1, b=2)` does not have the same cache entry as `foo(b=2, a=1)`.

Reply:
I'm going to merge the PR, it is becoming a blocker and given we cannot use the decorator directly because of the `dict` args this point can be resolved.